### ISSUE TITLE
File-scope unused variable not detected when ?> is present

### ISSUE THREAD

**Issue by GaryJones** [sirbrillig/phpcs-variable-analysis#232]
Snippet 1:

```php
<?php

$unused_variable = 'foo';
```

With the default configuration (`phpcs --standard=VariableAnalysis -s phpcs-test.php`), this is detected as "Unused variable $unused_variable (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)". ✅ 

Snippet 2:

```php
<?php

$unused_variable = 'foo';
?>
```

The addition of the `?>` stops the violation from being reported, even though the variable is still unused. ❌ 

Tested in VA 2.11.0.

**Comment by sirbrillig** [comment]
That definitely looks like a bug. I hadn't considered files that use closing php tags (since they're generally discouraged) and I suspect that the sniff misidentifies them. We'll need to make it aware of that.

**Comment by GaryJones** [comment]
Consider the case of a template file which is most likely to contain file-scope variables, that may include many `?>` and `<?php` as it breaks from PHP to HTML and back, then accounting for the `?>` seems like an important thing :-)

**Comment by sirbrillig** [comment]
100% agree. I'm honestly surprised this hasn't been brought up before! I'm working on a fix.
